jwst/datamodels/extension.py

```diff
@@ -0,0 +1,30 @@
+import os.path
+from asdf.extension import AsdfExtension
+from asdf import util
+
+SCHEMA_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), 'schemas'))
+
+URL_PREFIX = 'http://jwst.stsci.edu/schemas/'
+
+
+class BaseExtension(AsdfExtension):
+    """
+    This asdf extension provides url mapping for the asdf resolver
+
+    This is needed so that the asdf resolver can find the datamodel schemas
+    in the jwst package.  This base extension needs to be imported in the
+    jwst package __init__ and have an entry_points entry in setup.py
+    """
+    @property
+    def types(self):
+        return []
+
+    @property
+    def tag_mapping(self):
+        return []
+
+    @property
+    def url_mapping(self):
+        return [(URL_PREFIX,
+                 util.filepath_to_url(SCHEMA_PATH) + '/{url_suffix}')]
```

Comment:
No, see my comment in the original issue here

https://github.com/spacetelescope/jwst/issues/3224#issuecomment-470293796

There's an inconsistency in how this extension was implemented 3 years ago vs how the transforms extension was.  It would be nice to resolve it, but it will break backwards compatibility.

Reply:
From what I wrote in the issue:

One little wrinkle is that we have 2 entry points for ASDF extensions in this package:

- `JWSTExtension` which makes the `astropy.modeling` transforms defined in this package available to `gwcs`, `asdf` and other packages
- `BaseExtension` which makes the datamodel schemas available in the same way

But the URLs for these 2 ASDF extensions are fundamentally different in format:

`JWSTExtension` maps `http://stsci.edu/schemas/jwst_pipeline/` to transform schemas
`BaseExtension` maps `http://jwst.stsci.edu/schemas/` to datamodel schemas

Further, whether the `.yaml` ending is included varies as well.  It must be included on `BaseExtension`, but it cannot be included on `JWSTExtension`.

And neither of these extension names indicate that they are for transforms schemas or datamodel schemas, something a name should probably do.

It would be nice to resolve these for consistency sake, though not necessarily in this PR.  Thoughts on the best way to resolve this?

Reply:
Just a note for future reference but the consistent and correct thing to do would be to add the `.yaml` here rather than removing it in the other extension. This is because the mapping's output should be a real location on disk, whereas the input URI is just a schema id which knows nothing about the file type.

Reply:
Agree.  I will add this to the issue about future changes along these lines.  But doing so will require us renaming references in all our schemas, so a large change.